test/e2e-helm/e2e_helm_olm_test.go

```diff
@@ -15,25 +15,88 @@
 
 	. "github.com/onsi/ginkgo" //nolint:golint
+	. "github.com/onsi/gomega" //nolint:golint
 
 	testutils "github.com/operator-framework/operator-sdk/test/internal"
 )
 
-var _ = PDescribe("Integrating Helm Projects with OLM", func() {
+var _ = Describe("Integrating Helm Projects with OLM", func() {
 	Context("with operator-sdk", func() {
+		const operatorVersion = "0.0.1"
+
 		BeforeEach(func() {
-			By("Turning off interactive prompts for all generation tasks.")
+			By("turning off interactive prompts for all generation tasks.")
 			replace := "operator-sdk generate kustomize manifests"
 			testutils.ReplaceInFile(filepath.Join(tc.Dir, "Makefile"), replace, replace+" --interactive=false")
 		})
 
 		AfterEach(func() {
+			By("destroying the deployed package manifests-formatted operator")
+			cleanupPkgManCmd := exec.Command(tc.BinaryName, "cleanup", "packagemanifests",
+				"--operator-version", operatorVersion,
+				"--timeout", "4m")
+			_, _ = tc.Run(cleanupPkgManCmd)
+
+			By("uninstalling CRD's")
+			_ = tc.Make("uninstall")
 		})
 
-		It("Should allow generate the OLM bundle and run it", func() {
+		It("should generate and run a valid OLM bundle and packagemanifests", func() {
+			By("building the bundle")
+			err := tc.Make("bundle")
+			Expect(err).NotTo(HaveOccurred())
+
+			By("validating the bundle")
+			bundleValidateCmd := exec.Command(tc.BinaryName, "bundle", "validate", "bundle")
+			_, err = tc.Run(bundleValidateCmd)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("building the operator bundle image")
+			// Use the existing image tag but with a "-bundle" suffix.
+			imageSplit := strings.SplitN(tc.ImageName, ":", 2)
+			bundleImage := path.Join("quay.io", imageSplit[0]+"-bundle")
+			if len(imageSplit) == 2 {
+				bundleImage += ":" + imageSplit[1]
+			}
+			err = tc.Make("bundle-build", "BUNDLE_IMG="+bundleImage)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("loading the project image into Kind cluster")
+			err = tc.LoadImageToKindClusterWithName(bundleImage)
+			Expect(err).Should(Succeed())
+
+			By("adding the 'packagemanifests' rule to the Makefile")
+			err = tc.AddPackagemanifestsTarget()
+			Expect(err).Should(Succeed())
+
+			By("generating the operator package manifests")
+			err = tc.Make("packagemanifests")
+			Expect(err).NotTo(HaveOccurred())
+
+			By("updating clusterserviceversion with the manager image")
+			testutils.ReplaceInFile(
+				filepath.Join(tc.Dir, "packagemanifests", operatorVersion,
+					fmt.Sprintf("e2e-%s.clusterserviceversion.yaml", tc.TestSuffix)),
+				"controller:latest", tc.ImageName)
+
+			By("installing crds to run packagemanifests")
+			err = tc.Make("install")
+			Expect(err).NotTo(HaveOccurred())
+
+			By("running the package")
+			runPkgManCmd := exec.Command(tc.BinaryName, "run", "packagemanifests",
+				"--install-mode", "AllNamespaces",
+				"--operator-version", operatorVersion,
+				"--timeout", "4m")
+			_, err = tc.Run(runPkgManCmd)
+			Expect(err).NotTo(HaveOccurred())
```

Comment:
Can we separate the `generate/run bundle` tests from the `generate/run packagemanifests` tests?

Can be done in a follow-up